Firestore/Source/Remote/FSTDatastore.m

```diff
@@ -475,18 +523,20 @@ - (GRPCCall *)createRPCWithRequestsWriter:(GRXWriter *)requestsWriter {
   @throw FSTAbstractMethodException();  // NOLINT
 }
 
-- (void)start {
+- (void)start:(id)delegate {
   [self.workerDispatchQueue verifyIsCurrentQueue];
 
   if (self.state == FSTStreamStateError) {
-    [self performBackoff];
+    [self performBackoff:delegate];
     return;
   }
 
   FSTLog(@"%@ %p start", NSStringFromClass([self class]), (__bridge void *)self);
   FSTAssert(self.state == FSTStreamStateInitial, @"Already started");
 
   self.state = FSTStreamStateAuth;
+  FSTAssert(_delegate == nil, @"Delegate must be nil");
```

Comment:
This message doesn't help understand the invariant you're trying to enforce. Maybe "Delegate must not be reassigned"?

Reply:
Changed it to "closeWithFinalState should only be called for a started stream that has an active delegate."